Tools/GenerateWebsiteData.ps1

```diff
@@ -1,74 +1,264 @@
+            if ($docParam -eq $null) {
+                # Parameters found in the parameter set but not in the actual cmdlet are common
+                # parameters, like Verbose, ErrorAction, etc. Ignore.
+                continue
+            }
+            if (($docParam | Get-Member "parameterSet") -eq $null) {
+                $docParam | Add-Member "parameterSet" @()
+            }
+            $docParam.parameterSet += $parameterSet.HackName
         }
     }
 
-    # Sanity check the data.
-    $warning = "";
-    if ($cloudProduct -eq "") {
-        $warning = "Cmdlet $($cmdlet.Name) is not associated with a product."
+    # Go through each syntax object and mark which parameter set it belongs to, and if it is the
+    # default parameter set.
+    ForEach ($syntaxObj in $docObj.syntax.syntaxItem) {
+        $matchingParameterSet = $null
+
+        # The syntax matches the parameter set IFF all syntax obj params are in the parameter set.
+        ForEach ($parameterSet in $parameterSets) {
+            $parameterSetMatches = $true
+
+            # Note that the syntax obj only contains defined prameters, not "common" parameters
+            # -Verbose. So we loop through the syntax objs' parameters, not the parameter set's.
+            ForEach ($syntaxParam in $syntaxObj.parameter) {
+                # Remove a property which is redundant.
+                if (($syntaxParam | Get-Member "description") -ne $null) {
+                    $syntaxParam.description = $null
+                }
+
+                # Is the Parameter Set parameter found?
+                $paramInBoth = ($parameterSet.Parameters | Where Name -eq $syntaxParam.Name) -ne $null
+                if (-Not $paramInBoth) {
+                    $parameterSetMatches = $false
+                    break
+                }
+            }
+
+            # Do the reverse lookup and confirm that all required parameters of the parameter of
+            # the parameter set are honored. It is possible to have one parameter set be a proper
+            # subset of another.
+            ForEach ($paramSetParam in $parameterSet.Parameters) {
+                if (-Not $paramSetParam.IsMandatory) {
+                    continue
+                }
+                $paramInBoth = ($syntaxObj.parameter | Where Name -eq $paramSetParam.Name) -ne $null
+                if (-Not $paramInBoth) {
+                    $parameterSetMatches = $false
+                    break
+                }
+            }
+
+            if ($parameterSetMatches) {
+                $matchingParameterSet = $parameterSet
+                break
+            }
+        }
+
+
+        if ($matchingParameterSet -eq $null) {
+            Write-Warning "Unable to map syntax to parameter set for $cmdletName"
+            exit
+        }
+
+        Write-Host "`tMarking syntax for parameter set $($matchingParameterSet.HackName)"
+        $syntaxObj | Add-Member "parameterSet" $matchingParameterSet.HackName
```

Comment:
/shrug, I'm inclined to leave it as is.

If we were to define a constant we would only use it in one place. So it would make things slightly more complicated since it would separate definition from usage. Putting all the definitions in the same place would make them easier to update, but it's unlikely that we would want to rename the "parameterSet" field name.